### ISSUE TITLE
Error on applying casted renamed procedure

### ISSUE THREAD

**Issue by philnguyen** [racket/typed-racket#676]
### What version of Racket are you using?
`v6.12.0.3`

### What program did you run?
```racket
#lang typed/racket/base

(define (id x) x)
(define id1 (procedure-rename id 'identity))
(define id2 (cast id1 (Any -> Any)))
(id2 42)
```

### What should have happened?
Program runs

### If you got an error message, please include it here.
```
; g10: arity mismatch;
;  the expected number of arguments does not match the given number
;   given: 1
;   arguments...:
;    42
```

**Comment by AlexKnauth** [comment]
The issue is that `procedure-rename` only puts out `Procedure` as the output, so TR doesn't know that its safe to call `id1` with anything. If you called `(id1 42)` you would get a type error; the `cast` transforms that _would-be_-type-error into a runtime error. We could improve this by making the type of `procedure-rename` something like:
```
(All (A ... B)
  (-> (-> A ... A B) Symbol (-> A ... A B)))
```
Or, since that doesn't allow things like keyword functions or multi-valued functions, we could make a special case for typechecking `procedure-rename` so that it produces the same function type.

**Comment by Syntacticlosure** [comment]
I think this type for procedure-rename could solve the problem:
(All (A) (-> (Intersection A Procedure) Symbol A))
But I don't know why it doesn't work in TR....

**Comment by capfredf** [comment]
fixed by 47a5ab3e2f335e6956aea4b98700d22a359ad6b2
